engine/src/main/java/io/zeebe/engine/processor/workflow/job/DefaultJobCommandProcessor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.job;
+
+import io.zeebe.engine.processor.CommandProcessor;
+import io.zeebe.engine.processor.TypedRecord;
+import io.zeebe.engine.state.instance.JobState;
+import io.zeebe.engine.state.instance.JobState.State;
+import io.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.zeebe.protocol.record.RejectionType;
+import java.util.function.BiConsumer;
+
+/**
+ * Default implementation to process JobCommands to reduce duplication in CommandProcessor
+ * implementations.
+ */
+final class DefaultJobCommandProcessor<J extends JobRecord> implements CommandProcessor<J> {
+
+  private static final String NO_JOB_FOUND_MESSAGE =
+      "Expected to throw an error for job with key '%d', but no such job was found";
```

Comment:
I think `throw an error for job` this part is also dynamic, like `fail` and `complete`